src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -120,7 +120,14 @@ public void Add(MessageEndpointMapping mapping)
         /// </summary>
         protected override void BaseAdd(ConfigurationElement element)
         {
-            BaseAdd(element, true);
+            try
+            {
+                BaseAdd(element, true);
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                throw new Exception($"An ambigious message endpoint mapping has been defined at line: {e.Line}. Check the 'MessageEndpointMappings' section in {e.Filename} for conflicting mappings.", e);
```

Comment:
@timbussmann I was referring [to this comment](https://github.com/Particular/NServiceBus/pull/4141#r79126184)

My point was that the whole configuration system throws `ConfigurationErrorsException` that is why I was questioning the value of the generic exception wrapping and figured we should wrap it in that specific exception which the subsystem already provides. But I'm not hung up on that proposal